e2e/helper/verify/hedera.go

```diff
@@ -0,0 +1,579 @@
+
+	res, err := hedera.NewAccountAllowanceApproveTransaction().
+		ApproveTokenNftAllowance(
+			nftId,
+			ownerAccountId,
+			spenderAccountId,
+		).Execute(hederaClient)
+
+	if err != nil {
+		return nil, err
+	}
+
+	rec, err := res.GetReceipt(hederaClient)
+	if err != nil {
+		return nil, err
+	}
+	fmt.Printf("TX broadcasted. ID [%s], Status: [%s]\n", res.TransactionID, rec.Status)
+
+	time.Sleep(4 * time.Second)
+
+	return &res, err
+}
+
+func TopicMessagesWithStartTime(t *testing.T, hederaClient *hedera.Client, topicId hedera.TopicID, expectedValidatorsCount int, txId string, startTime int64) []string {
+	ethSignaturesCollected := 0
+	var receivedSignatures []string
+	signatureChannel := make(chan string)
+
+	fmt.Printf("Waiting for Signatures & TX Hash to be published to Topic [%v]\n", topicId.String())
+
+	// Subscribe to Topic
+	subscription, err := hedera.NewTopicMessageQuery().
+		SetStartTime(time.Unix(0, startTime)).
+		SetTopicID(topicId).
+		Subscribe(
+			hederaClient,
+			func(response hedera.TopicMessage) {
+				msg := &model.TopicMessage{}
+				err := proto.Unmarshal(response.Contents, msg)
+				if err != nil {
+					t.Fatal(err)
+				}
+
+				var transferID string
+				var signature string
+				switch msg.Message.(type) {
+				case *model.TopicMessage_FungibleSignatureMessage:
+					message := msg.GetFungibleSignatureMessage()
+					transferID = message.TransferID
+					signature = message.Signature
+					break
+				case *model.TopicMessage_NftSignatureMessage:
+					message := msg.GetNftSignatureMessage()
+					transferID = message.TransferID
+					signature = message.Signature
+				}
+
+				//Verify that all the submitted messages have signed the same transaction
+				if transferID != txId {
+					fmt.Printf("Expected signature message to contain the transaction id: [%s]\n", txId)
+				} else {
+					signatureChannel <- signature
+					fmt.Printf("Received Auth Signature [%s]\n", signature)
+				}
+			},
+		)
+	if err != nil {
+		t.Fatalf("Unable to subscribe to Topic [%s]", topicId)
+	}
+
+	timeoutTimer := time.NewTimer(120 * time.Second)
+
+signatureLoop:
+	for ethSignaturesCollected < expectedValidatorsCount {
+		select {
+		case signature := <-signatureChannel:
+			receivedSignatures = append(receivedSignatures, signature)
+			ethSignaturesCollected++
+		case <-timeoutTimer.C:
+			break signatureLoop
+		}
+	}
+
+	if ethSignaturesCollected != expectedValidatorsCount {
+		t.Fatalf("Expected the count of collected signatures to equal the number of validators: [%v], but was: [%v]", expectedValidatorsCount, ethSignaturesCollected)
+	}
+
+	subscription.Unsubscribe()
+
+	return receivedSignatures
+}
+
+func ScheduledNftAllowanceApprove(t *testing.T, hederaClient *hedera.Client, mirrorNodeClient *mirror_node.Client, bridgeAccount hedera.AccountID, payerAccount hedera.AccountID, expectedTransactionID string, startTimestamp int64) (transactionID, scheduleID string) {
```

Comment:
I think that we can remove `bridgeAccount` - currently, we are not using it in this function